### PR TITLE
CMake/MinGW fixes

### PR DESCRIPTION
**Pull request by carmethene**
Fixes a number of issues preventing compilation and linking via CMake when using the MinGW toolchain.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -46,6 +46,10 @@ CHECK_INCLUDE_FILE("alloca.h" HAVE_ALLOCA_H)
 #CHECK_INCLUDE_FILE("getopt.h" HAVE_GETOPT_H)
 CHECK_INCLUDE_FILE("malloc.h" HAVE_MALLOC_H)
 CHECK_INCLUDE_FILE("pcre2.h" HAVE_PCRE2_H)
+include(CheckSymbolExists)
+check_symbol_exists("memmem" "string.h" HAVE_MEMMEM)
+check_symbol_exists("strcasestr" "string.h" HAVE_MEMMEM)
```
Comment (rurban) [inline CMakeLists.txt:51]: Copy pasta

**Review by rurban — CHANGES_REQUESTED** [review]
One small error

File: CMakeLists.txt (hunk 1 above)
Comment (carmethene) [reply CMakeLists.txt]: Oops.

Comment (carmethene) [reply CMakeLists.txt]: I remade the commit and force pushed my branch, rather than polluting history with bad changes.  Hope that is acceptable.

Comment (rurban) [reply CMakeLists.txt]: Sure, force push is always preferred

**Review by rurban — APPROVED** [review]

**Review by rurban — APPROVED** [review]

**Pull request merged**